### PR TITLE
implement contributor name authority mappings

### LINKED ISSUE #1406: Map name authority, authorityURI, and valueURI attributes
For druid:cz548fn3478, expected:
```
<name type="corporate" usage="primary" authority="naf" authorityURI="http://id.loc.gov/authorities/names" valueURI="http://id.loc.gov/authorities/names/n82220006">
        <namePart>Stanford University. School of Engineering</namePart>
        <role>
          <roleTerm type="code" authority="marcrelator" authorityURI="http://id.loc.gov/vocabulary/relators" valueURI="http://id.loc.gov/vocabulary/relators/spn">spn</roleTerm>
          <roleTerm type="text" authority="marcrelator" authorityURI="http://id.loc.gov/vocabulary/relators" valueURI="http://id.loc.gov/vocabulary/relators/spn">Sponsor</roleTerm>
        </role>
      </name>
```

received:
```
<name type="corporate" usage="primary">
    <namePart>Stanford University. School of Engineering</namePart>
    <role>
      <roleTerm type="text" valueURI="http://id.loc.gov/vocabulary/relators/spn" authority="marcrelator" authorityURI="http://id.loc.gov/vocabulary/relators">Sponsor</roleTerm>
    </role>
  </name>
```

Note the missing name authority, authorityURI, and valueURI attributes.

### PR DESCRIPTION
**Pull request by ndushay**
## Why was this change made?

Fixes #1406 - maps MODS name authority attributes to cocina and back again

## How was this change tested?

unit test added

### Validations

1. For the record called out in #1406, cz548fn3478, I did a roundtrip validation for master branch and my branch.  Here is the diff:

```
$ diff cz548fn3478.master.txt cz548fn3478.mybranch.txt 
84c84
<   <name type="corporate" usage="primary">
---
>   <name type="corporate" usage="primary" valueURI="http://id.loc.gov/authorities/names/n82220006" authority="naf" authorityURI="http://id.loc.gov/authorities/names">
157c157,162
<           "value": "Stanford University. School of Engineering"
---
>           "value": "Stanford University. School of Engineering",
>           "uri": "http://id.loc.gov/authorities/names/n82220006",
>           "source": {
>             "code": "naf",
>             "uri": "http://id.loc.gov/authorities/names"
>           }
450c455
<   <name type="corporate" usage="primary">
---
>   <name type="corporate" usage="primary" valueURI="http://id.loc.gov/authorities/names/n82220006" authority="naf" authorityURI="http://id.loc.gov/authorities/names">
521c526
< <name type="corporate" usage="primary">
---
> <name type="corporate" usage="primary" valueURI="http://id.loc.gov/authorities/names/n82220006" authority="naf" authorityURI="http://id.loc.gov/authorities/names">
545c550
< <name type="corporate" usage="primary">
---
> <name type="corporate" usage="primary" valueURI="http://id.loc.gov/authorities/names/n82220006" authority="naf" authorityURI="http://id.loc.gov/authorities/names">
629c634
< <name type="corporate" usage="primary">
---
> <name type="corporate" usage="primary" valueURI="http://id.loc.gov/authorities/names/n82220006" authority="naf" authorityURI="http://id.loc.gov/authorities/names">
```

2.ran 500,000 records through round trip validation and it didn't get worse: `bin/validate-cocina-roundtrip -s 500000`

my branch:
```
To Cocina error: 57 of 500000 (0.0114%)
Data error: 17606 of 500000 (3.5212%)
Missing: 2954 of 500000 (0.5908%)

Error:  isn't one of in #/components/schemas/DescriptiveBasicValue/allOf/2/properties/value (57 errors)
Data error: [DATA ERROR] Subject has unknown authority code (14566 errors)
Data error: [DATA ERROR] originInfo/dateOther missing eventType (1019 errors)
Data error: [DATA ERROR] Subject contains a <name> element without a type attribute (660 errors)
Data error: [DATA ERROR] Contributor type incorrectly capitalized (473 errors)
Data error: [DATA ERROR] name/role/roleTerm missing value (457 errors)
Data error: Missing title (146 errors)
Data error: [DATA ERROR] Empty title node (90 errors)
Data error: [DATA ERROR] Name not found for title group (74 errors)
Data error: [DATA ERROR] name/namePart missing value (64 errors)
Data error: [DATA ERROR] Subject has <name> with an invalid type attribute 'Corporate' (26 errors)
Data error: [DATA ERROR] Contributor role code is missing authority (14 errors)
Data error: [DATA ERROR] Subject has <name> with an invalid type attribute '#N/A' (9 errors)
Data error: [DATA ERROR] Subject has <name> with an invalid type attribute 'Personal' (3 errors)
Data error: [DATA ERROR] Contributor type unrecognized 'pesonal' (3 errors)
Data error: [DATA ERROR] Unexpected node type for subject: 'cartographic' (1 errors)
Data error: [DATA ERROR] Subject has <name> with an invalid type attribute 'naf' (1 errors)
```

master branch:
```
To Cocina error: 57 of 500000 (0.0114%)
Data error: 17606 of 500000 (3.5212%)
Missing: 2954 of 500000 (0.5908%)

Error:  isn't one of in #/components/schemas/DescriptiveBasicValue/allOf/2/properties/value (57 errors)
Data error: [DATA ERROR] Subject has unknown authority code (14566 errors)
Data error: [DATA ERROR] originInfo/dateOther missing eventType (1019 errors)
Data error: [DATA ERROR] Subject contains a <name> element without a type attribute (660 errors)
Data error: [DATA ERROR] Contributor type incorrectly capitalized (473 errors)
Data error: [DATA ERROR] name/role/roleTerm missing value (457 errors)
Data error: Missing title (146 errors)
Data error: [DATA ERROR] Empty title node (90 errors)
Data error: [DATA ERROR] Name not found for title group (74 errors)
Data error: [DATA ERROR] name/namePart missing value (64 errors)
Data error: [DATA ERROR] Subject has <name> with an invalid type attribute 'Corporate' (26 errors)
Data error: [DATA ERROR] Contributor role code is missing authority (14 errors)
Data error: [DATA ERROR] Subject has <name> with an invalid type attribute '#N/A' (9 errors)
Data error: [DATA ERROR] Subject has <name> with an invalid type attribute 'Personal' (3 errors)
Data error: [DATA ERROR] Contributor type unrecognized 'pesonal' (3 errors)
Data error: [DATA ERROR] Unexpected node type for subject: 'cartographic' (1 errors)
Data error: [DATA ERROR] Subject has <name> with an invalid type attribute 'naf' (1 errors)
```

### CODE REVIEW

File: app/services/cocina/from_fedora/descriptive/contributor.rb [hunk 1]
```diff
@@ -96,6 +97,26 @@ def build_contributor_hash(name)
           end
         end
 
+        def name_authority_uri(name_el, contributor_hash)
+          value_uri = name_el.xpath('@valueURI', mods: DESC_METADATA_NS).first
+          if value_uri&.content
+            name_hash = contributor_hash[:name].first
+            name_hash[:uri] = value_uri&.content
```
Comment (jcoyne) [inline app/services/cocina/from_fedora/descriptive/contributor.rb:104]: hasn't value_uri.content already been determined to be not nil?

File: app/services/cocina/from_fedora/descriptive/contributor.rb [hunk 2]
```diff
@@ -96,6 +97,26 @@ def build_contributor_hash(name)
...
+          end
+          contributor_hash
+        end
+
+        def name_authority_source(code, uri)
+          source = {
+            code: code,
+            uri: uri
+          }.compact
+          source if source.presence
```
Comment (jcoyne) [inline app/services/cocina/from_fedora/descriptive/contributor.rb:117]:
```suggestion
          source.presence
```

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: app/services/cocina/from_fedora/descriptive/contributor.rb (hunk 1 above)
Comment (ndushay) [reply app/services/cocina/from_fedora/descriptive/contributor.rb:104]: yes!  whoops.

File: app/services/cocina/from_fedora/descriptive/contributor.rb (hunk 2 above)
Comment (ndushay) [reply app/services/cocina/from_fedora/descriptive/contributor.rb:117]: cool - I didn't even know about this method -- rubocop told me about it.

**Comment by ndushay** [comment]
@jcoyne suggestions taken - thanks!   Feel free to re-review.

**Review by jcoyne — APPROVED** [review]

**Pull request merged**